geode-core/src/test/java/org/apache/geode/cache/query/internal/AttributeDescriptorTest.java

```diff
@@ -232,18 +307,91 @@ public void readReflectionForImplicitMethodInvocationShouldThrowExceptionWhenMet
     AttributeDescriptor attributeDescriptor =
         new AttributeDescriptor(typeRegistry, methodInvocationAuthorizer, attributeName);
 
-    assertThatThrownBy(() -> attributeDescriptor.readReflection(testBean))
+    assertThatThrownBy(() -> attributeDescriptor.readReflection(testBean, queryExecutionContext))
+        .isInstanceOf(NotAuthorizedException.class)
+        .hasMessageStartingWith(RestrictedMethodAuthorizer.UNAUTHORIZED_STRING);
+  }
+
+  @Test
+  @Parameters({
+      "nonPublicAttributeWithPublicAccessor, nonPublicAttributeWithPublicAccessor",
+      "nonPublicAttributeWithPublicGetterMethod, getNonPublicAttributeWithPublicGetterMethod"})
+  public void readReflectionForImplicitMethodInvocationShouldShouldThrowExceptionWhenMethodIsNotAuthorizedAndCacheResult(
+      String attributeName, String methodName) {
+    doReturn(false).when(methodInvocationAuthorizer).authorize(any(), any());
+    AttributeDescriptor attributeDescriptor =
+        new AttributeDescriptor(typeRegistry, methodInvocationAuthorizer, attributeName);
+
+    assertThatThrownBy(() -> attributeDescriptor.readReflection(testBean, queryExecutionContext))
         .isInstanceOf(NotAuthorizedException.class)
         .hasMessageStartingWith(RestrictedMethodAuthorizer.UNAUTHORIZED_STRING);
+
+    String cacheKey = TestBean.class.getCanonicalName() + "." + methodName;
+    assertThat(queryExecutionContext.cacheGet(cacheKey)).isNotNull();
```

Comment:
See earlier comment regarding whether we need to assert that the value is not null *and* that it's equal to false.

Reply:
👍